src/service/viewer-impl.js

```diff
@@ -499,6 +511,34 @@ export class Viewer {
     return this.isWebviewEmbedded_;
   }
 
+  /**
+   * Whether the document is using Chrome Custom Tabs.
+   * @return {boolean}
+   */
+  isCustomTab() {
+    return this.isCustomTab_;
+  }
+
+  /**
+   * Update the URL fragment with data needed to support custom tabs.
+   */
+  maybeUpdateFragmentForCustomTab() {
+    if (!this.isCustomTab()) {
+      return;
+    }
+    if (!this.win.history.replaceState) {
+      return;
+    }
+    // Ensure that canonical URL origin matches document URL.
```

Comment:
Not sure I understand this. Don't we want this to work for CDN-served AMP pages?